dashboard/app/controllers/courses_controller.rb

```diff
@@ -3,9 +3,16 @@ class CoursesController < ApplicationController
 
   before_action :require_levelbuilder_mode, except: [:index, :show, :vocab, :resources, :code, :standards]
   before_action :authenticate_user!, except: [:index, :show, :vocab, :resources, :code, :standards]
+  before_action :set_unit_group, only: [:show, :vocab, :resources, :code, :standards, :edit, :update, :get_rollup_resources]
   before_action :set_redirect_override, only: [:show]
   authorize_resource class: 'UnitGroup', except: [:index]
 
+  def set_unit_group
+    @unit_group = UnitGroup.get_from_cache(params[:course_name])
```

Comment:
What happens if course_name isn't there?

Reply:
I'm seeing @unit_group defined here, and then also @unit_group defined below in `create`. How do these interact?

Reply:
Course name is part of the url so I don't think you can request the page without it?

Reply:
Oh interesting. Do you know if the URL gets set via the cache? Or vice versa: the cache gets set based on the URL?

Reply:
course_name is set up by the routes which for this are specified here: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/routes.rb#L281. More on the magic happening here: https://guides.rubyonrails.org/routing.html. Then its using that course_name to find the correct UnitGroup from the cache with `get_from_cache` which falls back on `find_by` if it can't find it

Reply:
Wow, thanks for the explanation––very cool

Reply:
> I'm seeing @unit_group defined here, and then also @unit_group defined below in create. How do these interact?

@megcrenshaw to answer your question a different way, `set_unit_group` is configured on line 6 to not be called on create, so in a very literal sense that is why those two codepaths do not interact.

Reply:
Ah, of course––thank you